pkg/api/pod/util_test.go

```diff
@@ -1187,50 +1191,52 @@ func TestDropProbeGracePeriod(t *testing.T) {
 		},
 	}
 
-	enabled := true
-	for _, oldPodInfo := range podInfo {
-		for _, newPodInfo := range podInfo {
-			oldPodHasGracePeriod, oldPod := oldPodInfo.hasGracePeriod, oldPodInfo.pod()
-			newPodHasGracePeriod, newPod := newPodInfo.hasGracePeriod, newPodInfo.pod()
-			if newPod == nil {
-				continue
-			}
-
-			t.Run(fmt.Sprintf("feature enabled=%v, old pod %v, new pod %v", enabled, oldPodInfo.description, newPodInfo.description), func(t *testing.T) {
-
-				var oldPodSpec *api.PodSpec
-				if oldPod != nil {
-					oldPodSpec = &oldPod.Spec
+	for _, enabled := range []bool{true, false} {
+		for _, oldPodInfo := range podInfo {
+			for _, newPodInfo := range podInfo {
+				oldPodHasGracePeriod, oldPod := oldPodInfo.hasGracePeriod, oldPodInfo.pod()
+				newPodHasGracePeriod, newPod := newPodInfo.hasGracePeriod, newPodInfo.pod()
+				if newPod == nil {
+					continue
 				}
-				dropDisabledFields(&newPod.Spec, nil, oldPodSpec, nil)
 
-				// old pod should never be changed
-				if !reflect.DeepEqual(oldPod, oldPodInfo.pod()) {
-					t.Errorf("old pod changed: %v", cmp.Diff(oldPod, oldPodInfo.pod()))
-				}
+				t.Run(fmt.Sprintf("feature enabled=%v, old pod %v, new pod %v", enabled, oldPodInfo.description, newPodInfo.description), func(t *testing.T) {
+					defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.ProbeTerminationGracePeriod, enabled)()
```

Comment:
enabled will change in the t.Run (will be closed over).  I think you need to have:

```
gateEnabled := enabled
t.Run(... func() {
  defer featuregatetesting.....(gateEnabled)
```

(the value of enabled will be incorrect when tests are accumulated and then run in parallel)

Reply:
This is copied from how `TestDropAppArmor` does it above, are those tests also wrong?

Reply:
referencing things outside the closure is fine as long as the test doesn't set t.Parallel() (if it did that, t.Run wouldn't block, and @smarterclayton would be correct... variables outside the closure would change before the test completed running)

Reply:
Nothing in this file appears to set `t.Parallel()`, so I think this should be okay?

Another example from above: https://github.com/kubernetes/kubernetes/blob/68dadd40d638c814d96977442751eaccd31d98c1/pkg/api/pod/util_test.go#L807-L817

Reply:
I'm fine leaving the subtle bug in here if we expect the test to catch it (since it's unlikely we'll make these parallel in the near term).